applications/FluidDynamicsApplication/custom_conditions/adjoint_monolithic_wall_condition.h

```diff
@@ -0,0 +1,418 @@
+
+    using VectorType = Vector;
+
+    using MatrixType = Matrix;
+
+    using IndexType = std::size_t;
+
+    using SizeType = std::size_t;
+
+    using EquationIdVectorType = std::vector<std::size_t>;
+
+    using DofsVectorType = std::vector<Dof<double>::Pointer>;
+
+    using DofsArrayType = PointerVectorSet<Dof<double>, IndexedObject>;
+
+    static constexpr IndexType TBlockSize = (TDim + 1);
+
+    static constexpr IndexType TFluidLocalSize = TBlockSize * TNumNodes;
+
+    static constexpr IndexType TCoordsLocalSize = TDim * TNumNodes;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    AdjointMonolithicWallCondition(
+        IndexType NewId = 0) : Condition(NewId)
+    {
+    }
+
+    AdjointMonolithicWallCondition(
+        IndexType NewId,
+        const NodesArrayType& ThisNodes)
+        : Condition(NewId, ThisNodes)
+    {
+    }
+
+    AdjointMonolithicWallCondition(
+        IndexType NewId,
+        GeometryType::Pointer pGeometry)
+        : Condition(NewId, pGeometry)
+    {
+    }
+
+    AdjointMonolithicWallCondition(
+        IndexType NewId,
+        GeometryType::Pointer pGeometry,
+        PropertiesType::Pointer pProperties)
+        : Condition(NewId, pGeometry, pProperties)
+    {
+    }
+
+    AdjointMonolithicWallCondition(AdjointMonolithicWallCondition const& rOther)
+        : Condition(rOther)
+    {
+    }
+
+    ~AdjointMonolithicWallCondition() override = default;
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    /// Assignment operator
+    AdjointMonolithicWallCondition& operator=(AdjointMonolithicWallCondition const& rOther)
+    {
+        Condition::operator=(rOther);
+        return *this;
+    }
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    Condition::Pointer Create(
+        IndexType NewId,
+        NodesArrayType const& ThisNodes,
+        PropertiesType::Pointer pProperties) const override
+    {
+        return Kratos::make_intrusive<AdjointMonolithicWallCondition>(
+            NewId, GetGeometry().Create(ThisNodes), pProperties);
+    }
+
+    Condition::Pointer Create(
+        IndexType NewId,
+        GeometryType::Pointer pGeom,
+        PropertiesType::Pointer pProperties) const override
+    {
+        return Kratos::make_intrusive<AdjointMonolithicWallCondition>(NewId, pGeom, pProperties);
+    }
+
+    Condition::Pointer Clone(
+        IndexType NewId,
+        NodesArrayType const& rThisNodes) const override
+    {
+        Condition::Pointer pNewCondition =
+            Create(NewId, GetGeometry().Create(rThisNodes), pGetProperties());
+
+        pNewCondition->SetData(this->GetData());
+        pNewCondition->SetFlags(this->GetFlags());
+
+        return pNewCondition;
+    }
+
+    void Initialize(const ProcessInfo& rCurrentProcessInfo) override
+    {
+        this->SetValue(ADJOINT_EXTENSIONS, Kratos::make_shared<ThisExtensions>(this));
+    }
+
+    int Check(
+        const ProcessInfo& rCurrentProcessInfo) const override;
```

Comment:
```suggestion
    int Check(const ProcessInfo& rCurrentProcessInfo) const override;
```
Style guide (inline 1 argument function).

Reply:
done :)